### ISSUE TITLE
Tried to export project to snapshot

### ISSUE THREAD

**Issue by tobiasK2001** [Open-Systems-Pharmacology/PK-Sim#1779]
Any idea what could be the issue here?

Best, Tobias

Application:
PK-Sim® 9.1.2

Object reference not set to an instance of an object.

Stack trace:
```
at PKSim.Core.Snapshots.Mappers.CompoundPropertiesMapper.formulationSelectionFrom(FormulationMapping formulationMapping, PKSimProject project)
   at PKSim.Core.Snapshots.Mappers.CompoundPropertiesMapper.<>c__DisplayClass8_0.<formulationMappingFrom>b__0(FormulationMapping x)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at PKSim.Core.Snapshots.Mappers.CompoundPropertiesMapper.formulationMappingFrom(IReadOnlyList`1 formulationMappings, PKSimProject project)
   at PKSim.Core.Snapshots.Mappers.CompoundPropertiesMapper.protocolPropertiesFrom(ProtocolProperties protocolProperties, PKSimProject project)
   at PKSim.Core.Snapshots.Mappers.CompoundPropertiesMapper.<>c__DisplayClass4_0.<MapToSnapshot>b__0(CompoundProperties x)
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBase`2.SnapshotFrom(TModel model, Action`1 configurationAction)
   at PKSim.Core.Snapshots.Mappers.CompoundPropertiesMapper.<MapToSnapshot>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBaseExtensions.<MapTo>d__9`2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SimulationMapper.<MapToSnapshot>d__22.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBaseExtensions.<MapTo>d__9`2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.ProjectMapper.<mapSimulationsToSnapshots>d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.ProjectMapper.<MapToSnapshot>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBase`2.<MapToSnapshot>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Services.SnapshotTask.<exportSnapshotFor>d__21`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at PKSim.Core.Snapshots.Services.SnapshotTask.<ExportModelToSnapshot>d__6`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at OSPSuite.Core.Extensions.TaskExtensions.<SecureAwait>d__2`1.MoveNext()
```

**Comment by tobiasK2001** [comment]
can it have something to do that I loaded some fromulation BB in developer mode and now I am trying to export in "normal mode"?

**Comment by msevestre** [comment]
@tobiasK2001 
It looks like it indeed....the call stack says formulation Selection. Probably what you loaded from snapshot in dev mode is somewhat corrupted. Did you edit the snapshot by hand before importing?

**Comment by msevestre** [comment]
Ifbyou

**Comment by tobiasK2001** [comment]
Yes I did modify the formulation snapshot by hand before importing. I think I added the batch-name for the formulation used for the different particle bins. I am not sure if I really need these imported formulations... Maybe I can just delete them and export without them
Project is for client and confidential, so I will try if I can solve it here.
Thanks for your help!
Best, Tobias

**Comment by tobiasK2001** [comment]
I think I solved the issue: To check if it is an formulation issue I exported each formulation or administration protokoll as snapshot in developer mode. This worked. So no issue there.
Than to further explore I exported each Simulation: this did not work for 4 Simulations (out of 60 or so). These Simulations were imported from other projects via the Working-journal. Actually these were DDI reference simulations I just needed to test the DDI compound template performance. These simulations were created from DDI simulation snapshots or just taken from PK-sim template reference files. As I didn't need sem any more I deleted the four Simulations from the project. Than a Jason file could be produced without problems :-)
I am just loading the project from the snapshot again. I keep you posted, when other issues arise .

Best, Tobias
